### ISSUE TITLE
More descriptive error message when reading /etc/hosts file

### ISSUE THREAD

**Issue by hurufu** [skarnet/s6-dns#4]
Hello, thanks for writing and maintaining s6 suite of tools.

I had quite a head-scratcher recently, because it was hard to deduce why `s6-tcpclient` failed with error:
```
s6-tcpclient: fatal: unable to init DNS: Invalid or incomplete multi-byte or wide character
```
I had to resort to reading sources... The problem was that I had some strange entries in /etc/hosts file:

```
fe80::1%lo0 localhost
0.0.0.0 0.0.0.0
```

Is it possible to extend error messages to inform that at least it was an error related to /etc/hosts file parsing? Or maybe ignore erroneous lines, because this file is commonly managed manually, so it is possible that it will contain some errors in it.

**Comment by skarnet** [comment]
Yeah, that problem was also reported to me from another source. The problem is twofold:

- indeed `/etc/hosts` parsing should warn on unrecognized entries and go to the next line instead of failing altogether. I'll work on this for the next version of s6-dns.
- Also, s6-tcpclient should not unconditionally load `/etc/hosts`, I'll add an option to do that on demand. I'll add this to the next version of s6-networking.

Sorry I don't have a quick fix, apart from commenting the entries with expanded syntax in `/etc/hosts`.

**Comment by hurufu** [comment]
I don't need a quick fix, if it will be available in the next release – that's good enough.
